source/MaterialXGenShader/Nodes/ImageNode.cpp

```diff
@@ -0,0 +1,40 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenShader/Nodes/ImageNode.h>
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenShader/ShaderNode.h>
+#include <MaterialXGenShader/ShaderStage.h>
+#include <MaterialXGenShader/ShaderGenerator.h>
+#include <MaterialXGenShader/Util.h>
+
+namespace MaterialX
+{
+
+ShaderNodeImplPtr ImageNode::create()
+{
+    return std::make_shared<ImageNode>();
+}
+
+void ImageNode::addInputs(ShaderNode& node, GenContext&) const
+{
+    // Add additional scale and offset inputs
+    ShaderInput* input = node.addInput("uv_scale", Type::VECTOR2);
+    input->setValue(Value::createValue<Vector2>(Vector2(1.0f, 1.0f)));
+    input = node.addInput("uv_offset", Type::VECTOR2);
+    input->setValue(Value::createValue<Vector2>(Vector2(0.0f, 0.0f)));
+}
+
+void ImageNode::setValues(const Node& /*node*/, ShaderNode& /*shaderNode*/, GenContext& /*context*/) const
+{
+    // TODO: Share code with HWImageNode.
```

Comment:
Need to figure out how to share logic between Hw and non-hw.

Reply:
Do you really need this for the non-hw case? For OSL udim handling is built into the texture sampler code. So as far as I can see it's only the flip that you need and not the scale/offset in the non-hw case. So maybe the extra scale/offset inputs could only be used in the HW case where it's needed, and not here.

Reply:
I was thinking on this and was just trying to make it consistent but true that UDIM baking would probably never occur for s/w I'll just revert the OSL changes and remove this node variant.